src/library/scala/collection/Factory.scala

```diff
@@ -223,6 +223,21 @@ trait IterableFactory[+CC[_]] extends Serializable {
   def tabulate[A](n1: Int, n2: Int, n3: Int, n4: Int, n5: Int)(f: (Int, Int, Int, Int, Int) => A): CC[CC[CC[CC[CC[A]]]] @uncheckedVariance] =
     tabulate(n1)(i1 => tabulate(n2, n3, n4, n5)(f(i1, _, _, _, _)))
 
+  /** Concatenates all argument collections into a single $coll.
+   *
+   *  @param xss the collections that are to be concatenated.
+   *  @return the concatenation of all the collections.
+   */
+  def concat[A](xss: Iterable[A]*): CC[A] = {
+    val b = newBuilder[A]
+    // At present we're using IndexedSeq as a proxy for "has a cheap size method".
+    if (xss forall (_.isInstanceOf[IndexedSeq[_]]))
+      b.sizeHint(xss.map(_.size).sum)
+
+    for (xs <- xss) b ++= xs
+    b.result()
+  }
```

Comment:
We should use a view-based implementation. We can optimize it in `StrictOptimizedSeqFactory`, though.

Reply:
The view based implementation could look like:

~~~ scala
from(xss.foldLeft(View.empty[A])((v, xs) => v ++ xs))
~~~

Reply:
Ah ok. That sounds good. I was wondering if it was worth extending `View.Concat` to allow concatenating an arbitrary number of elements.

Reply:
I guess it makes the implementation a lot more complex though, so I'll stick with the `foldLeft` to construct the view.